qemu/tests/ansible_test.py

```diff
@@ -0,0 +1,83 @@
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Ansible playbook basic test:
+    1) Check ansible package exists
+    2) Launch the guest
+    3) Clone an ansible playbook repo
+    4) Generate the ansible-playbook command
+    5) Execute the playbook and verify the return status
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    sm = software_manager.SoftwareManager()
+    if not (sm.check_installed("ansible") or sm.install("ansible")):
+        test.cancel("ansible package install failed")
+
+    get_ip_timeout = params.get_numeric("get_ip_timeout")
+    guest_user = params["username"]
+    guest_passwd = params["password"]
+    ansible_callback_plugin = params["ansible_callback_plugin"]
+    ansible_addl_opts = params.get("ansible_addl_opts", "")
+    ansible_ssh_extra_args = params["ansible_ssh_extra_args"]
+    ansible_extra_vars = params.get("ansible_extra_vars", "{}")
+    playbook_repo = params["playbook_repo"]
+    playbook_timeout = params.get_numeric("playbook_timeout")
+    playbook_dir = params.get("playbook_dir",
+                              os.path.join(test.logdir, "ansible_playbook"))
+    tl_playbook = os.path.join(playbook_dir, params["tl_playbook"])
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    guest_ip = vm.wait_for_get_address(0, get_ip_timeout)
+
+    try:
+        logging.info("Clone %s", playbook_repo)
+        process.run("git clone {src} {dst}".format(src=playbook_repo,
+                                                   dst=playbook_dir),
+                    verbose=False)
+
+        error_context.base_context("Generate playbook related options",
+                                   logging.info)
+        extra_vars = {"ansible_ssh_extra_args": ansible_ssh_extra_args,
+                      "ansible_ssh_pass": guest_passwd}
+        extra_vars.update(json.loads(ansible_extra_vars))
+
+        ansible_cmd_options = ["ansible-playbook",
+                               "-u {}".format(guest_user),
+                               "-i {},".format(guest_ip),
+                               "-e '{}'".format(json.dumps(extra_vars)),
+                               ansible_addl_opts,
+                               tl_playbook]
+        ansible_cmd = r" ".join(ansible_cmd_options)
+
+        error_context.context("Execute ansible playbook", logging.info)
+        env_vars = {"ANSIBLE_STDOUT_CALLBACK": ansible_callback_plugin}
+        play_s, play_o = process.getstatusoutput(ansible_cmd,
+                                                 timeout=playbook_timeout,
+                                                 ignore_status=False,
+                                                 shell=False, env=env_vars)
+        logging.debug("Full ansible playbook output for debugging:\n%s",
+                      play_o)
```

Comment:
Will use `utils_misc.log_line` to handle it.

Reply:
Since `process.getstatusoutput` can't register a output handler in order to logging message with timestamp (neither other functions in `process`), suggest performing by `write` directly or alternatives.